### PR TITLE
#162414119 implement user authentication actions and reducers

### PR DESCRIPTION
**Pull request by iMichaelOwolabi**
#### What does this PR do?
Implement user authentication actions and reducers (Signup and Login)
#### Description of Task to be completed?
Implement authentication actions and reducers so that the components can render data from redux
#### How should this be manually tested?
pull the repository and cd into the branch and then run npm test
#### Any background context you want to provide?
This is required for the components to have access to data
#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/162414119 #162414119
#### Questions:

### CODE REVIEW

File: src/actions/userActions.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+export const signUpFailure = payload => ({
+  type: SIGNUP_FAILURE,
+  payload
+});
+
+export const userSignUp = user => async (dispatch) => {
+  try {
+    dispatch(authStart());
+    const response = await signupCall({ user });
```
Comment (codrex) [inline src/actions/userActions.js]:
```suggestion
    const response = await signupCall(user);
```

File: src/actions/userActions.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+export const loginFailure = payload => ({
+  type: LOGIN_FAILURE,
+  payload
+});
+
+export const userLogin = user => async (dispatch) => {
+  try {
+    dispatch(authStart());
+    const response = await loginCall({ user });
```
Comment (codrex) [inline src/actions/userActions.js]:
```suggestion
    const response = await loginCall(user );
```

File: src/actions/index.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import actionTypes from './actionTypes';
```
Comment (codrex) [inline src/actions/index.js]: This should be in its own file

**Review by codrex — APPROVED** [review]

File: src/actions/index.js (hunk 3 above)
Comment (iMichaelOwolabi) [reply src/actions/index.js]: Done

**Review by iAfolayan — APPROVED** [review]

File: src/actions/test/authActions.test.js [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+import * as actions from '../authActions';
+import actionTypes from '../actionTypes';
+import * as axios from '../../helpers/axiosHelper/auth';
+import {
+  payload,
+  signupOkResponse,
+  loginOkResponse,
+  mockResponse,
+  fakeUser2
+} from '../../../__mocks__';
```
Comment (codrex) [inline src/actions/test/authActions.test.js]: `__mocks__` is meant for jest and not for storing mock data for testing. Please make use of the mock data folder

File: src/reducers/authreducer.test.js [hunk 5]
```diff
@@ -0,0 +1,85 @@
+import actionTypes from '../actions/actionTypes';
+import initialState from './initialState';
+import authReducer from './authReducer';
+import {
+  loginOkResponse,
+  loginBadResponse,
+  signupOkResponse,
+  error,
+} from '../../__mocks__';
```
Comment (codrex) [inline src/reducers/authreducer.test.js]: ☝️

File: src/actions/test/authActions.test.js (hunk 4 above)
Comment (iMichaelOwolabi) [reply src/actions/test/authActions.test.js]: Ok. will fix

**Review by codrex — APPROVED** [review]

**Pull request merged**